### PR TITLE
check for MCLAG connections before deleting MCLAG domain

### PR DESCRIPTION
**Pull request by mrbojangles3**
This patch was attached to a fork that I deleted. We use branches here.

### CODE REVIEW

File: pkg/webhook/connection/connection_webhook.go [hunk 1]
```diff
@@ -142,6 +142,37 @@ func (w *Webhook) ValidateDelete(ctx context.Context, obj runtime.Object) (admis
...
+		}
+		if len(mgmtConnList.Items) <= 1 {
+			return nil, errors.Errorf("management connection is last one and cannot be deleted")
+		}
+	}
+
+	// This is a light check to make sure that no mclags links transit this domain before we delete the domain
+	if conn.Spec.MCLAGDomain != nil {
+		labels := conn.Spec.ConnectionLabels()
+		searchLabels := make(client.MatchingLabels)
```
Comment (Frostman) [inline pkg/webhook/connection/connection_webhook.go:160]: you don't need to create a copy as .ConnectionLabels() already returns a new map

File: pkg/webhook/connection/connection_webhook.go [hunk 2]
```diff
@@ -142,6 +142,37 @@ func (w *Webhook) ValidateDelete(ctx context.Context, obj runtime.Object) (admis
...
+		for key, val := range labels {
+			searchLabels[key] = val
+		}
+		// overwrite the MCLAG-Domain ConnectionType with just an MCLAG type
+		searchLabels[wiringapi.LabelConnectionType] = wiringapi.ConnectionTypeMCLAG
+		mclagList := &wiringapi.ConnectionList{}
+		// The matching here, will logically and the key/vals in searchLabels together
+		// giving just the relevant connections
+		if err := w.Client.List(ctx, mclagList, searchLabels); err != nil {
+			return nil, errors.Errorf("error listing servers on management connection")
```
Comment (Frostman) [inline pkg/webhook/connection/connection_webhook.go:170]: wrong error message - it's listing matching mclag connections

File: pkg/webhook/connection/connection_webhook.go [hunk 3]
```diff
@@ -142,6 +142,37 @@ func (w *Webhook) ValidateDelete(ctx context.Context, obj runtime.Object) (admis
 	if len(extAttachments.Items) > 0 {
 		return nil, errors.Errorf("connection has external attachments")
 	}
+	if conn.Spec.Management != nil {
+		mgmtConnList := &wiringapi.ConnectionList{}
+		if err := w.Client.List(ctx, mgmtConnList, client.MatchingLabels{
+			wiringapi.LabelConnectionType: wiringapi.ConnectionTypeManagement,
+		}); err != nil {
+			return nil, errors.Errorf("error listing servers on management connection")
```
Comment (Frostman) [inline pkg/webhook/connection/connection_webhook.go:150]: wrong error message - it's listing mgmt connections here

File: pkg/webhook/connection/connection_webhook.go [hunk 4]
```diff
@@ -142,6 +142,37 @@ func (w *Webhook) ValidateDelete(ctx context.Context, obj runtime.Object) (admis
...
+		// overwrite the MCLAG-Domain ConnectionType with just an MCLAG type
+		searchLabels[wiringapi.LabelConnectionType] = wiringapi.ConnectionTypeMCLAG
+		mclagList := &wiringapi.ConnectionList{}
+		// The matching here, will logically and the key/vals in searchLabels together
+		// giving just the relevant connections
+		if err := w.Client.List(ctx, mclagList, searchLabels); err != nil {
+			return nil, errors.Errorf("error listing servers on management connection")
+		}
+		if len(mclagList.Items) > 0 {
+			return nil, errors.Errorf("%d MCLAGS link(s) present. Delete those before this domain", len(mclagList.Items))
```
Comment (Frostman) [inline pkg/webhook/connection/connection_webhook.go:173]: I think `%d MCLAG connection(s) using this domain present, delete those before the domain` would be a bit more clear

**Review by Frostman — CHANGES_REQUESTED** [review]
just a few minor things

File: pkg/webhook/connection/connection_webhook.go (hunk 3 above)
Comment (mrbojangles3) [reply pkg/webhook/connection/connection_webhook.go]: will do

File: pkg/webhook/connection/connection_webhook.go (hunk 2 above)
Comment (mrbojangles3) [reply pkg/webhook/connection/connection_webhook.go]: will do

File: pkg/webhook/connection/connection_webhook.go (hunk 4 above)
Comment (mrbojangles3) [reply pkg/webhook/connection/connection_webhook.go]: will do

File: pkg/webhook/connection/connection_webhook.go (hunk 1 above)
Comment (mrbojangles3) [reply pkg/webhook/connection/connection_webhook.go]: Thank you

**Review by Frostman — APPROVED** [review]

**Pull request merged**